middlewares/user/getTargetUserIdByPMChannelId.js

```diff
@@ -0,0 +1,57 @@
+const {StreamChat} = require('stream-chat');
+const Getstream = require('../../vendor/getstream');
+
+/**
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {import('express').NextFunction} next
+ * @returns
+ */
+const getTargetUserIdByPMChannelIdMiddleware = async (req, res, next) => {
+  const {userId} = req || {};
+  const {channelId} = req?.query || {};
+
+  if (!channelId) {
+    return res.status(400).json({
+      code: 400,
+      success: false,
+      message: 'Channel ID is required'
+    });
+  }
+
+  const chatClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+  let channelMembers;
+  try {
+    let {users} = await Getstream.chat.getTargetUserIdByChannelId(chatClient, userId, channelId);
+    channelMembers = users;
+  } catch (e) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The StreamChat client is being instantiated with API_KEY and SECRET directly from the environment variables. It's better to validate these values before using them. Also, error handling in this section only logs the error message but does not provide any specific details about the error to the user. Consider improving the error message to make it more informative.
```diff
-  const chatClient = new StreamChat(process.env.API_KEY, process.env.SECRET);
+  const { API_KEY, SECRET } = process.env;
+  if (!API_KEY || !SECRET) {
+    return res.status(500).json({
+      code: 500,
+      success: false,
+      message: 'Missing necessary environment variables'
+    });
+  }
+  const chatClient = new StreamChat(API_KEY, SECRET);

   try {
     let {users} = await Getstream.chat.getTargetUserIdByChannelId(chatClient, userId, channelId);
     channelMembers = users;
   } catch (e) {
-    console.error('Failed to get channel members', e);
+    console.error('Failed to get channel members due to:', e.message);
     return res.status(500).json({
       code: 500,
       success: false,
-      message: 'Internal server error'
+      message: `Error occurred while fetching channel members: ${e.message}`
     });
   }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->